src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -325,10 +371,21 @@ function NewTransactionModal ({ handleClose, toggleModal }) {
   </Modal>
 }
 
-export function ConfirmationModal ({ transaction, handleClose, declineTransaction, payTransaction }) {
+export function ConfirmationModal ({ transaction, handleClose, declineTransaction, refundTransaction, payTransaction, setCounterpartyNotFound, counterpartyNotFound }) {
   const { newMessage } = useFlashMessageContext()
-  if (!transaction) return null
-  const { id, counterparty, amount, type, action } = transaction
+  const { id, amount, type, action } = transaction
+  const { counterparty = {} } = transaction
+  const { holofuelCounterparty, loading } = useCounterparty(counterparty.id)
+
+  useEffect(() => {
+    if (!transaction) return null
+    else if (holofuelCounterparty || loading) {
```

Comment:
the `loading` condition here doesn't do anything, because it will never be the case that both `loading` and `holofuelCounterparty.notFound` are true at the same time.

Reply:
Good catch!